client/client.go

```diff
@@ -87,6 +87,24 @@ type Config struct {
 	// Password is the password for the specified user to add as an authorization header
 	// to the request.
 	Password string
+
+	// TimeoutPerRequest specifies the time limit for a single request made by the
+	// Client. The timeout includes connection time, any redirects, and reading
+	// the response body. The timeout applies on most APIs except Watcher.Next(),
+	// because watch is supposed to wait for long time.
+	//
+	// One API call may send multiple requests to different etcd servers until it
+	// succeeds. Use context of the API to specify the overall timeout.
+	//
+	// A TimeoutPerRequest of zero means no timeout.
+	TimeoutPerRequest time.Duration
+
+	// HandshakeTimeoutPerWatchRequest specifies the time limit for handshake
+	// phase of a single watch request. The timeout includes connection time
+	// and acknowledge from server to indicate that watch is under way.
+	//
+	// A HandshakeTimeoutPerWatchRequest of zero means no timeout.
+	HandshakeTimeoutPerWatchRequest time.Duration
```

Comment:
That will greatly simplify this commit and make everything consistent and easy to understand.